### PR TITLE
Update SDL for v2

### PR DESCRIPTION
**Pull request by czoido**
Port SDL recipe to new tools and make it more compatible with v2

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]
Looks great and it's a nice example for me to follow in other recipes to help me port them :)

File: recipes/sdl/all/conanfile.py [hunk 1]
```diff
@@ -80,20 +84,24 @@ class SDLConan(ConanFile):
...
-    generators = ["cmake", "pkg_config"]
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def generate(self):
+        self.define_toolchain()
+        lib_paths = [lib for _, dep in self.dependencies.items() for lib in dep.cpp_info.libdirs]
```
[the remark this answers is not in the document]
Comment (czoido) [reply recipes/sdl/all/conanfile.py:92]: Let's fix this in the next iteration, thank you so much for the review!

File: recipes/sdl/all/conanfile.py [hunk 2]
```diff
@@ -80,20 +84,24 @@ class SDLConan(ConanFile):
...
+        env = Environment()
+        env.define_path("LIBRARY_PATH", os.pathsep.join(lib_paths))
+
+        # FIXME: remove and raise required_conan_version to 1.55 once it's on c3i
```
Comment (prince-chrismc) [inline recipes/sdl/all/conanfile.py:96]: Should be able to fix this

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**